### PR TITLE
Generator enhancements and bug fixes

### PR DESCRIPTION
**Pull request by 5A11**
## Proposed changes

- Changed encode and decode methods from class methods to static methods
- Added "_performative" to the end of performative names in protocol buffer schema file
- Added 'test_configuration_file_not_valid' to the generator cli tests 
- Addressed optional content erroneous is_set assertion

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/protocols/generator.py [hunk 1]
```diff
@@ -1185,8 +1182,8 @@ def _custom_types_module_str(self) -> str:
             cls_str += '        """\n'
             cls_str += "        raise NotImplementedError\n\n"
 
-            cls_str += "    @classmethod\n"
-            cls_str += '    def decode(cls, {}_protobuf_object) -> "{}":\n'.format(
+            cls_str += "    @staticmethod\n"
```
Comment (DavidMinarsch) [inline aea/protocols/generator.py]: Decode is a classmethod, only encode is not :)

File: tests/data/generator/t_protocol/t_protocol.proto [hunk 2]
```diff
@@ -98,12 +98,12 @@ message TProtocolMessage{
...
     int32 target = 4;
     oneof performative{
-        Performative_Ct performative_ct = 5;
-        Performative_Empty_Contents performative_empty_contents = 6;
-        Performative_Mt performative_mt = 7;
-        Performative_O performative_o = 8;
-        Performative_Pct performative_pct = 9;
-        Performative_Pmt performative_pmt = 10;
-        Performative_Pt performative_pt = 11;
+        Performative_Ct_performative performative_ct = 5;
```
Comment (DavidMinarsch) [inline tests/data/generator/t_protocol/t_protocol.proto]: Seems like we should have `_Performative` to have consistent capitalization?

File: aea/protocols/generator.py (hunk 1 above)
Comment (5A11) [reply aea/protocols/generator.py]: Done

File: tests/data/generator/t_protocol/t_protocol.proto (hunk 2 above)
Comment (5A11) [reply tests/data/generator/t_protocol/t_protocol.proto]: Done

**Pull request merged**